apis/python/src/tiledbsoma/io/ingest.py

```diff
@@ -89,6 +90,16 @@
 _TDBO = TypeVar("_TDBO", bound=TileDBObject[RawHandle])
 
 
+AdditionalMetadata = Optional[Dict[str, str]]
+
+
+def add_metadata(
+    obj: TileDBObject[Any], additional_metadata: AdditionalMetadata
+) -> None:
+    for k, v in (additional_metadata or {}).items():
```

Comment:
check it's open for write and throw an informative error message if it isn't

I believe there's code for that already in `ingest.py` -- either as an already-extracted helper function, or inline code that should be extracted to a helper function